core/src/main/java/greencity/controller/FactOfTheDayController.java

```diff
@@ -63,9 +70,138 @@
+                                                                               Pageable pageable) {
         return ResponseEntity.status(HttpStatus.OK)
             .body(factOfTheDayService.getAllFactsOfTheDay(pageable));
     }
+
+    /**
+     * Method which return {@link FactOfTheDayDTO} by given id.
+     *
+     * @param id of {@link FactOfTheDay}
+     * @return {@link FactOfTheDayDTO}
+     */
+    @ApiOperation(value = "Find fact of the day by given id.")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = HttpStatuses.OK, response = FactOfTheDayDTO.class),
+        @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN),
+    })
+    @GetMapping("/find")
+    public ResponseEntity<FactOfTheDayDTO> findFactOfTheDay(@ApiIgnore @AuthenticationPrincipal
+                                                               Principal principal, @RequestParam("id") Long id) {
+        return ResponseEntity.status(HttpStatus.OK)
+            .body(factOfTheDayService.getFactOfTheDayById(id));
+    }
+
+    /**
+     * Method which saves {@link FactOfTheDay}.
+     *
+     * @param factOfTheDayPostDTO of {@link FactOfTheDayPostDTO}
+     * @return {@link ErrorsResponseDto} with of operation and errors fields
+     */
+    @ApiOperation(value = "Save fact of the day")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = HttpStatuses.OK, response = ErrorsResponseDto.class),
+        @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN),
+    })
+    @ResponseBody
+    @PostMapping("/")
+    public ErrorsResponseDto saveFactOfTheDay(@Valid @RequestBody FactOfTheDayPostDTO factOfTheDayPostDTO,
+                                              BindingResult bindingResult) {
+        if (bindingResult.hasErrors()) {
+            ErrorsResponseDto errorsResponseDto = new ErrorsResponseDto();
+            errorsResponseDto.setStatus(false);
+            for (FieldError fieldError : bindingResult.getFieldErrors()) {
+                errorsResponseDto.getErrors().add(
+                    new FieldErrorDto(fieldError.getField(), fieldError.getDefaultMessage()));
+            }
+            return errorsResponseDto;
+        }
+        factOfTheDayService.saveFactOfTheDayAndTranslations(factOfTheDayPostDTO);
+        return new ErrorsResponseDto(true, new ArrayList<>());
+    }
+
+    /**
+     * Method which updates {@link FactOfTheDay}.
+     *
+     * @param factOfTheDayPostDTO of {@link FactOfTheDayPostDTO}
+     * @return {@link ErrorsResponseDto} with of operation and errors fields
+     */
+    @ApiOperation(value = "Update fact of the day")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = HttpStatuses.OK),
+        @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN)
+    })
+    @ResponseBody
+    @PutMapping("/")
+    public ErrorsResponseDto updateFactOfTheDay(@ApiIgnore @AuthenticationPrincipal
```

Comment:
`ErrorsResponseDto` name is confusing. Looking at the code it seems that whenever is done inside the method, it will return an error. Maybe rename something like `GenericResponseDto`.

Reply:
Okay